pipelines/controller/src/main/java/org/openmrs/analytics/DWHFilesManager.java

```
@@ -0,0 +1,353 @@
+        deleteDirectoryAndFiles(path);
+      }
+    }
+  }
+
+  /**
+   * This method deletes all the files, directories and subdirectories under the given root
+   * directory. This is done by first deleting all the files under the root directory including
+   * under the subdirectories. Later the directories and the subdirectories are deleted.
+   *
+   * @param rootDirectory which needs to be deleted
+   * @throws IOException
+   */
+  public void deleteDirectoryAndFiles(ResourceId rootDirectory) throws IOException {
+    ResourceId filesResourceToBeMatched =
+        rootDirectory.resolve("**", StandardResolveOptions.RESOLVE_FILE);
+    List<MatchResult> matchedFilesResultList =
+        FileSystems.matchResources(Collections.singletonList(filesResourceToBeMatched));
+
+    // Collect the matched files which also includes the files under the subdirectories, at the same
+    // time collect the directories as well.
+    List<ResourceId> fileResources = new ArrayList<>();
+    Set<String> allDirectoryPathSet = new HashSet<>();
+    for (MatchResult matchResult : matchedFilesResultList) {
+      if (matchResult.status() == Status.OK && !matchResult.metadata().isEmpty()) {
+        for (Metadata metadata : matchResult.metadata()) {
+          fileResources.add(metadata.resourceId());
+          ResourceId currentDirectory = metadata.resourceId().getCurrentDirectory();
+          allDirectoryPathSet.add(currentDirectory.toString());
+          // Beam currently creates a temporary empty folder with name '.temp-beam' sometimes under
+          // the directories. These empty subdirectories could not be determined with the existing
+          // Beam file system apis, hence this is hard-coded here so that these empty directories
+          // and their parent directories can be successfully deleted.
+          allDirectoryPathSet.add(
+              currentDirectory
+                  .resolve(".temp-beam", StandardResolveOptions.RESOLVE_DIRECTORY)
```

What happens if `.temp-beam` does not exist in `currentDirectory`?

---

If there is no `.temp-beam` directory in the `currentDirectory` then the `FileSystems.delete(..)` api ignores them (internally the beam is catching and ignoring the exception, check [here](https://github.com/apache/beam/blob/26f5792c8cc7f0e2a04a3b94c57a96feabf92156/sdks/java/core/src/main/java/org/apache/beam/sdk/io/LocalFileSystem.java#L206)).

---

Great, thanks for checking.